### PR TITLE
Update docs to make Predictive behavior more clear

### PR DESCRIPTION
**Pull request by tomwallis**
This PR addresses [this issue](https://github.com/pyro-ppl/numpyro/issues/1845#issuecomment-2284569263), improving the docs for how the `Predictive` class is expected to function. In particular, I try to make it more clear that the observations should be set to `None` to explicitly make prior or posterior predictions. 

I updated the docstring for the class itself, and also two of the example `ipynb`s.

In the docstring files there are also some `black` import re-orderings.

### CODE REVIEW

**Review by fehiepsi — APPROVED** [review]
Thanks, @tomwallis!

File: numpyro/infer/util.py [hunk 1]
```diff
@@ -1,19 +1,18 @@
...
 from contextlib import contextmanager
 from functools import partial
 from typing import Callable, Optional
-import warnings
-
-import numpy as np
 
 import jax
+import jax.numpy as jnp
+import numpy as np
```
Comment (fehiepsi) [inline numpyro/infer/util.py:12]: Could you revert the order of those imports? This seems to make CI failing.

**Comment by fehiepsi** [comment]
@tomwallis We are going to make a new patch release soon. Do you want to incorporate this change in the release?

**Comment by tomwallis** [comment]
Sorry @fehiepsi I was on holiday then too busy. Hopefully this fixes the issue.

**Pull request merged**